src/main/java/org/ays/auth/repository/AysUserRepository.java

```diff
@@ -19,6 +21,15 @@ public interface AysUserRepository extends JpaRepository<AysUserEntity, String>,
      */
     Optional<AysUserEntity> findByEmailAddress(String emailAddress);
 
+    /**
+     * Finds a user by their phone number, which is a concatenation of country code and line number.
+     *
+     * @param phoneNumber the concatenated phone number (country code + line number) of the user to be found
+     * @return an optional containing the UserEntity with the given phone number, or an empty optional if not found
+     */
+    @Query("SELECT u FROM AysUserEntity u WHERE CONCAT(u.countryCode, u.lineNumber) = :phoneNumber")
+    Optional<AysUserEntity> findByPhoneNumber(@Param("phoneNumber") String phoneNumber);
```

Comment:
Aşağıdaki gibi yaparsak concat yapmamıza ve query kullanmamıza ihtiyaç kalmamış olur
```suggestion
    Optional<AysUserEntity> findByCountryCodeAndLineNumber(String countryCode, String lineNumber);
```

Reply:
değiştirdim